### PR TITLE
specs for string_ext.rb

### PR DESCRIPTION
**Pull request by flash1452**
For #200

### CODE REVIEW

File: spec/ssh_scan/string_ext_spec.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  context "when unhexing a string" do
+    testing_sample = "48656C6C6F20576F726C6421"
+    it "should load all the attributes properly" do
+      expected_unhexedstr = "Hello World!"
+      test_result = testing_sample.unhexify
+      expect(test_result).to eql(expected_unhexedstr)
+    end
+  end
+
+	context "when hexing a string" do
```
Comment (claudijd) [inline spec/ssh_scan/string_ext_spec.rb]: Some tabbing/spacing issues here.

File: spec/ssh_scan/string_ext_spec.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		end
+	end
+
+	context "when resolving a DNS name as IPv4" do
+	testing_dns = "192.168.1.1"
+		it "should load all the attributes properly" do
+		  test_result = testing_dns.resolve_fqdn_as_ipv4.to_s
+		  expect(test_result).to eql("192.168.1.1")
+		end
+	end
```
Comment (claudijd) [inline spec/ssh_scan/string_ext_spec.rb]: This test is failing in the build

Comment (claudijd) [inline spec/ssh_scan/string_ext_spec.rb]: maybe we check github.com and change the test expectation to match on an ip regex?

**Comment by claudijd** [comment]
Thanks for the PR, added a couple things we'd like sorted before this lands.  Overall, great job!

File: spec/ssh_scan/string_ext_spec.rb (hunk 1 above)
Comment (flash1452) [inline spec/ssh_scan/string_ext_spec.rb]: Fixed

**Pull request merged**